r/configure

```diff
@@ -25,6 +25,9 @@
 # INCLUDE_DIR and LIB_DIR manually via e.g:
 # R CMD INSTALL --configure-vars='INCLUDE_DIR=/.../include LIB_DIR=/.../lib'
 
+# Fail script on failing commands
+set -e
```

Comment:
I think this is what's causing the [Travis failure](https://travis-ci.org/apache/arrow/jobs/545216104#L740-L741) on the job that build the package without libarrow. pkg-config exists but it isn't finding libarrow, as expected, but the configure script is aborting early. I suspect that if you remove this, we'll get a passing build, and then this can be merged.